src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -519,7 +521,7 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
             ExecutionStrategyParameters newParameters = parameters.transform(builder ->
                     builder.executionStepInfo(stepInfoForListElement)
                             .nonNullFieldValidator(nonNullableFieldValidator)
-                            .listSize(values.size())
+                            .listSize(size.orElse(-1)) // -1 signals that we don't know the size
```

Comment:
I had a look at `graphql.execution.ExecutionStrategyParameters#getListSize` - we never use it.

I was worried this `-1` semantics could affect some one )badly but we never use it - I think this will be ok

Reply:
yes. Maybe we can introduce a new method that returns OptionalInt ?